tests/kafkatest/tests/connect/connect_test.py

```diff
@@ -100,14 +100,12 @@ def test_file_source_and_sink(self, converter="org.apache.kafka.connect.json.Jso
         self.zk.start()
         self.kafka.start()
 
-        source_external_props = os.path.join(self.source.PERSISTENT_ROOT, "connect-file-external.properties")
-        self.source.node.account.create_file(source_external_props, self.render('connect-file-external.properties'))
         self.source.set_configs(lambda node: self.render("connect-standalone.properties", node=node), [self.render("connect-file-source.properties")])
-
-        sink_external_props = os.path.join(self.sink.PERSISTENT_ROOT, "connect-file-external.properties")
-        self.sink.node.account.create_file(sink_external_props, self.render('connect-file-external.properties'))
         self.sink.set_configs(lambda node: self.render("connect-standalone.properties", node=node), [self.render("connect-file-sink.properties")])
 
+        self.source.set_external_configs(lambda node: self.render("connect-file-external.properties", node=node))
```

Comment:
One thing that confuses me a little bit is the dependence on the name "connect-file-external.properties." It is hard-coded above as `EXTERNAL_FILE`, but then why do we need it again here? I'm actually wondering if the full lambda generalization is needed. Could this instead be a flag which simply indicates whether rendering of the external config is desired or not? Or perhaps you had additional use cases in mind?

Reply:
The files happen to be the same, but this actually refers to the test-specific template whereas `EXTERNAL_FILE` refers to the generated file used by the connector. I'll rename the test-specific template file so it's less confusing.

Reply:
To be honest, I used a lambda to follow the same pattern as the existing `set_configs` method.